src/scalaDebugger.ts

```diff
@@ -46,15 +50,169 @@ export async function start(
+      })
+      .then((result) => {
+        if (result === mainClassPick) {
+          return this.provideDebugMainClassConfiguration().then((config) => {
+            if (config === undefined) return [];
+            return [config];
+          });
+        } else if (result === testClassPick) {
+          return this.provideDebugTestClassConfiguration().then((config) => {
+            if (config === undefined) return [];
+            return [config];
+          });
+        }
+        return [];
+      });
   }
 
   resolveDebugConfiguration(
     _folder: WorkspaceFolder | undefined,
     debugConfiguration: DebugConfiguration
   ): ProviderResult<DebugConfiguration> {
+    if (debugConfiguration.type === undefined) return null;
     return debugConfiguration;
   }
+
+  private provideDebugMainClassConfiguration(): Thenable<
+    DebugConfiguration | undefined
+  > {
+    return this.askForBuildTarget().then((buildTarget) => {
+      if (buildTarget === undefined) return undefined;
+      return this.askForClassName().then((className) => {
+        if (className === undefined) return undefined;
+        return this.askForArguments().then((args) => {
+          if (args === undefined) return undefined;
+          return this.askForConfigurationName(className).then((name) => {
+            if (name === undefined) return undefined;
+            const result: DebugConfiguration = {
+              type: configurationType,
+              name: name,
+              request: launchRequestType,
+              mainClass: className,
+              buildTarget: buildTarget,
+              args: args,
+            };
+            return result;
+          });
+        });
+      });
+    });
+  }
+
+  private provideDebugTestClassConfiguration(): Thenable<
+    DebugConfiguration | undefined
+  > {
+    return this.askForBuildTarget().then((buildTarget) => {
+      if (buildTarget === undefined) return undefined;
+      return this.askForClassName().then((className) => {
+        if (className === undefined) return undefined;
+        return this.askForConfigurationName(className).then((name) => {
+          if (name === undefined) return undefined;
+          const result: DebugConfiguration = {
+            type: configurationType,
+            name: name,
+            request: launchRequestType,
+            testClass: className,
+            buildTarget: buildTarget,
+          };
+          return result;
+        });
+      });
+    });
+  }
+
+  private askForBuildTarget(): Thenable<string | null | undefined> {
+    return vscode.window
+      .showInputBox({
+        prompt: "Enter name of the build target",
+        placeHolder: "Optional, you can leave it empty",
+      })
+      .then((buildTarget) => {
+        if (buildTarget === undefined) {
+          return undefined;
+        } else if (buildTarget === "") {
+          return null;
+        } else {
+          return buildTarget;
+        }
+      });
+  }
```

Comment:
Can we use just `string | undefined` instead of carrying also `null` as a third option?

Reply:
now with `reject` we can :) previously there was collision -- we had to carry information, if user rejected input, and if they just provided no build target.